addons/api/addon/serializers/credential-library.js

```diff
@@ -31,9 +32,39 @@ export default class CredentialLibrarySerializer extends ApplicationSerializer {
         return super.serialize(...arguments);
     }
   }
+  serializeAttribute(snapshot, json) {
+    const value = super.serializeAttribute(...arguments);
+    const { credential_type, type, credential_mapping_overrides, isNew } =
+      snapshot.record;
+    if (
+      type === TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC &&
+      credential_mapping_overrides &&
+      credential_type &&
+      !isNew
+    ) {
+      // API expects to send null to fields if it is undefined or deleted
+      const remainingAttrs = Object.entries(
+        options.mapping_overrides[credential_type],
+      ).reduce(
+        (obj, [key]) =>
+          Object.assign(
+            obj,
+            credential_mapping_overrides[key] ? {} : { [key]: null },
+          ),
+        {},
+      );
+
+      json.credential_mapping_overrides = {
+        ...credential_mapping_overrides,
+        ...remainingAttrs,
+      };
```

Comment:
Few comments regarding this, I don't think you needed to use `Object.entries` since you only used the keys anyways. And couldn't we have included the actual values of the fields as part of the reduce rather than keeping them separate and doing something like this:

```suggestion
      json.credential_mapping_overrides = Object.keys(
        options.mapping_overrides[credential_type],
      ).reduce((obj, key) => {
        if (credential_mapping_overrides[key]) {
          obj[key] = credential_mapping_overrides[key];
        } else {
          obj[key] = null;
        }
        return obj;
      }, {});
```